src/openrct2-ui/interface/Graph.cpp

```diff
@@ -182,10 +183,129 @@ static void graph_draw_line_b_money32(
     }
 }
 
+struct financial_tooltip_info
+{
+    const CoordsXY coords;
+    const money32 money{};
+};
+
+struct Rect
+{
+    const CoordsXY LeftTop;
+    const CoordsXY RightBottom;
+
+    int32_t GetLeft() const
+    {
+        return LeftTop.x;
+    }
+    int32_t GetTop() const
+    {
+        return LeftTop.y;
+    }
+    int32_t GetRight() const
+    {
+        return RightBottom.x;
+    }
+    int32_t GetBottom() const
+    {
+        return RightBottom.y;
+    }
+    int32_t GetWidth() const
+    {
+        return RightBottom.x - LeftTop.x;
+    }
+    int32_t GetHeight() const
+    {
+        return RightBottom.y - LeftTop.y;
+    }
+    bool Contains(const CoordsXY& coords) const
+    {
+        return coords.x >= LeftTop.x && coords.x <= RightBottom.x && coords.y >= LeftTop.y && coords.y <= RightBottom.y;
```

Comment:
I'd use `GetLeft`, `GetRight`, etc instead of the hardcoded internal variables, but that's my personal preference.